### PR TITLE
Feature/reports

### PR DESCRIPTION
**Pull request by Zablas**

### CODE REVIEW

File: Backend/Models/Configurations/ReportTypeConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                Enum.GetValues(typeof(ReportTypeId))
+                    .Cast<ReportTypeId>()
+                    .Select(rt => new ReportType
+                    {
+                        Id = rt,
+                        Name = rt.ToString()
+                    })
+            );
+
+
```
Comment (wizacass) [inline Backend/Models/Configurations/ReportTypeConfiguration.cs:24]: Extra newline

File: Backend/Models/ReportEntity/Report.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+        [Required]
+        public ReportTypeId ReportTypeId { get; set; }
+
+        [Required]
+        public ReportType ReportType { get; set; }
+
+        public int UserId { get; set; }
+
+        public User User { get; set; }
```
Comment (Anonimas1) [inline Backend/Models/ReportEntity/Report.cs:56]: User  should be required

File: Backend/Models/ReportEntity/Report.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        public int PharmacyId { get; set; }
+
+        public Pharmacy Pharmacy { get; set; }
+
```
Comment (Anonimas1) [inline Backend/Models/ReportEntity/Report.cs:61]: Pharmacy should be required

File: Backend/Models/ReportEntity/Report.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        [Required]
+        public ReportType ReportType { get; set; }
+
+        public int UserId { get; set; }
+
+        public User User { get; set; }
+
+        public int PharmacyId { get; set; }
+
+        public Pharmacy Pharmacy { get; set; }
```
Comment (wizacass) [inline Backend/Models/ReportEntity/Report.cs:60]: Are user and Pharmacy not required?

**Review by wizacass — CHANGES_REQUESTED** [review]

File: Backend/Models/Configurations/ReportTypeConfiguration.cs (hunk 1 above)
Comment (Zablas) [reply Backend/Models/Configurations/ReportTypeConfiguration.cs]: Fixed

File: Backend/Models/ReportEntity/Report.cs (hunk 4 above)
Comment (Zablas) [reply Backend/Models/ReportEntity/Report.cs]: Fixed

File: Backend/Models/ReportEntity/Report.cs (hunk 2 above)
Comment (Zablas) [reply Backend/Models/ReportEntity/Report.cs]: Fixed

File: Backend/Models/ReportEntity/Report.cs (hunk 3 above)
Comment (Zablas) [reply Backend/Models/ReportEntity/Report.cs]: Fixed

**Review by Anonimas1 — APPROVED** [review]

**Review by wizacass — APPROVED** [review]

**Pull request merged**